### PR TITLE
Invite functionality dev

### PR DESCRIPTION
**Pull request by Exino329**
Created the invite functionality in the overview. The user can click on the Join button and a dialog will appear. Then the user can enter the code (which is the id of the trips he wants to join that is send by someone).
-  If the code is not a trip ID or if the trip ID inserted is already a trip that belongs to the user, displaying an error message.
-  If the code is valid, add the Trips for the user and display it in the overView.

### CODE REVIEW

File: app/src/main/java/com/github/se/wanderpals/model/viewmodel/OverviewViewModel.kt [hunk 1]
```diff
@@ -41,4 +42,25 @@ open class OverviewViewModel(val tripsRepository: TripsRepository) : ViewModel()
...
   }
+  /**
+   * Adds the user to a trip with the specified trip ID. The trip must already exist in the database
+   * to make this function success.
+   *
+   * @param tripId The ID of the trip to join.
+   * @return True if the user successfully joins the trip, false otherwise.
+   */
+  open fun joinTrip(tripId: String): Boolean {
+    var success = false
```
Comment (Polo150202) [inline app/src/main/java/com/github/se/wanderpals/model/viewmodel/OverviewViewModel.kt:53]: Maybe you can use lateinit in place of double initialisation

**Review by Polo150202 — APPROVED** [review]
The code and the tests looks pretty good, some details to fix but nothing big

File: app/src/main/java/com/github/se/wanderpals/model/viewmodel/OverviewViewModel.kt (hunk 1 above)
Comment (Exino329) [reply app/src/main/java/com/github/se/wanderpals/model/viewmodel/OverviewViewModel.kt:53]: I think that it is globably safer to let the success being false, so we can capture the logic that before trying the trip ID there is no success

File: app/src/main/java/com/github/se/wanderpals/ui/screens/overview/Overview.kt [hunk 2]
```diff
@@ -68,3 +95,91 @@ fun Overview(overviewViewModel: OverviewViewModel, navigationActions: Navigation
...
+ * Composable function to handle a dialog for inserting a trip code.
+ *
+ * @param closeDialogueAction A lambda function to be called when the dialog is dismissed.
+ * @param addTripCodeAction A lambda function that takes a trip code String as input and returns a
+ *   Boolean indicating whether the trip code was successfully added.
+ */
+@Composable
+fun DialogHandler(closeDialogueAction: () -> Unit, addTripCodeAction: (String) -> Boolean) {
+
+  val EMPTY_CODE = ""
```
[the remark this answers is not in the document]
Comment (Exino329) [reply app/src/main/java/com/github/se/wanderpals/ui/screens/overview/Overview.kt:109]: According to Kotlin documentation for const values, i think it is the correct naming convention : 

https://kotlinlang.org/docs/coding-conventions.html#choose-good-names

[the remark this answers is not in the document]
Comment (Exino329) [reply app/src/main/java/com/github/se/wanderpals/ui/screens/overview/Overview.kt:109]: I think we interpreted it differently, Anyway we can still resolve this in a future branch for fixing the syntax / documentation for every classes.

**Pull request merged**